R-package/tests/helper_scripts/generate_models.R

```diff
@@ -40,7 +40,7 @@ generate_regression_model <- function() {
   print('Regression')
   y <- rnorm(metadata$kRows)
 
-  data <- xgb.DMatrix(X, label = y)
+  data <- xgb.DMatrix(X, label = y, nthread = 1)
```

Comment:
I didn't think this file is used in tests.

Reply:
Is the file used for something outside of the R package tests, or can it be deleted? I couldn't find any usage anywhere, and the tests still pass if it gets deleted.

Reply:
The model is generated once, then future tests use the generated models for testing compatibility.

Reply:
Then I guess it doesn't hurt to limit threads there too, considering that the booster object also stores number of threads.